packages/components/icon/src/Icon.tsx

```diff
@@ -0,0 +1,142 @@
+import { css, cx } from 'emotion';
+import React, { forwardRef } from 'react';
+import tokens from '@contentful/f36-tokens';
+import { Primitive } from '@contentful/f36-core';
+import type {
+  CommonProps,
+  ComponentVariant,
+  Simplify,
+} from '@contentful/f36-core';
+import type {
+  ComponentType,
+  ExoticComponent,
+  ReactElement,
+  SVGAttributes,
+} from 'react';
+
+export type IconComponent = ExoticComponent<any> | ComponentType<any>;
+
+type IconVariant = Simplify<ComponentVariant | 'muted' | 'white'>;
+
+type AsOrChildren =
+  | {
+      as: ExoticComponent<any> | ComponentType<any>;
+      children?: never;
+    }
+  | {
+      as?: never;
+      children: ReactElement | ReactElement[];
+    };
+
+type IconInternalProps = {
+  /**
+   * The name prop is used to check if "trimmed" styling should be applied,
+   * which sets the `width` to `auto`.
+   */
+  name?: string;
+  /**
+   * Determines the size of the icon
+   */
+  size?: 'large' | 'medium' | 'small' | 'tiny';
+  /**
+   * Determines the fill color used
+   */
+  variant?: IconVariant;
+  /**
+   * SVG viewBox attribute
+   */
+  viewBox?: `${number} ${number} ${number} ${number}`;
+} & AsOrChildren;
+
+export type IconProps = IconInternalProps &
```

Comment:
I'd expect this component have MarginProps as Box component

Reply:
Shouldn't we add this to `<Primitive />` then? Are there components that shouldn't take those props?